rocksdb/columnfamily.nim

```diff
@@ -19,43 +19,48 @@
 
 {.push raises: [].}
 
-import
-  ./rocksdb
+import ./columnfamily/cfhandle, ./rocksdb
 
 export rocksdb
 
 type
   ColFamilyReadOnly* = object
     db: RocksDbReadOnlyRef
-    name: string
+    cfHandle: ColFamilyHandleRef
 
   ColFamilyReadWrite* = object
     db: RocksDbReadWriteRef
-    name: string
+    cfHandle: ColFamilyHandleRef
 
 proc withColFamily*(
-    db: RocksDbReadOnlyRef,
-    name: string): RocksDBResult[ColFamilyReadOnly] =
+    db: RocksDbReadOnlyRef, name: string
+): RocksDBResult[ColFamilyReadOnly] =
   ## Creates a new `ColFamilyReadOnly` from the given `RocksDbReadOnlyRef` and
   ## column family name.
+  doAssert not db.isClosed()
```

Comment:
in the functions that return a `Result` already, I wonder if it's worth adding this check and return an `err` if the db is closed..

Reply:
The argument for it being an assert is that it would be a programmer error if we try to operate on a closed database. At runtime the user of the API can call isClosed() to check if the database is closed if needed

Reply:
ok - let's leave it at that then